tests/benchmark/test_bloatnet.py

```diff
@@ -0,0 +1,618 @@
+    gas_costs = fork.gas_costs()
+
+    # this is only used for computing the intinsic gas
+    data = final_storage_value.to_bytes(32, "big").rstrip(b"\x00")
+
+    storage = Storage()
+
+    # Initial gas for PUSH0 + CALLDATALOAD + POP (at the end)
+    totalgas = gas_costs.G_BASE * 2 + gas_costs.G_VERY_LOW
+    totalgas = totalgas + fork.transaction_intrinsic_cost_calculator()(calldata=data)
+    gas_increment = gas_costs.G_VERY_LOW * 2 + gas_costs.G_STORAGE_SET + gas_costs.G_COLD_SLOAD
+    sstore_code = Op.PUSH0 + Op.CALLDATALOAD
+    storage_slot: int = 0
+    while totalgas + gas_increment < gas_benchmark_value:
+        totalgas += gas_increment
+        sstore_code = sstore_code + Op.SSTORE(storage_slot, Op.DUP1)
+        storage[storage_slot] = final_storage_value
+        storage_slot += 1
+
+    sstore_code = sstore_code + Op.POP  # Drop last value on the stack
+
+    sender = pre.fund_eoa()
+    print(sender)
+    contract_address = pre.deploy_contract(
+        code=sstore_code,
+        storage=Storage(),
+    )
+
+    tx_0_1 = Transaction(
+        to=contract_address,
+        gas_limit=gas_benchmark_value,
+        data=(final_storage_value // 2).to_bytes(32, "big").rstrip(b"\x00"),
+        value=0,
+        sender=sender,
+    )
+    tx_1_2 = Transaction(
+        to=contract_address,
+        gas_limit=gas_benchmark_value,
+        data=final_storage_value.to_bytes(32, "big").rstrip(b"\x00"),
+        value=0,
+        sender=sender,
+    )
+
+    post = {contract_address: Account(storage=storage)}
+
+    blockchain_test(pre=pre, blocks=[Block(txs=[tx_0_1, tx_1_2])], post=post)
+
+
+# Warm reads are very cheap, which means you can really fill a block
+# with them. Only fill the block by a factor of SPEEDUP.
+SPEEDUP: int = 100
+
+
+@pytest.mark.valid_from("Prague")
+def test_bloatnet_sload_warm(
+    blockchain_test: BlockchainTestFiller, pre: Alloc, fork: Fork, gas_benchmark_value: int
+):
+    """Test that loads warm storage locations many times."""
+    gas_costs = fork.gas_costs()
+
+    # Pre-fill storage with values
+    num_slots = 100  # Number of storage slots to warm up
+    storage = Storage({HashInt(i): HashInt(0xDEADBEEF + i) for i in range(num_slots)})
+
+    # Calculate gas costs
+    totalgas = fork.transaction_intrinsic_cost_calculator()(calldata=b"")
+
+    # First pass - warm up all slots (cold access)
+    warmup_gas = num_slots * (gas_costs.G_COLD_SLOAD + gas_costs.G_BASE)
+    totalgas += warmup_gas
+
+    # Calculate how many warm loads we can fit
+    gas_increment = gas_costs.G_WARM_SLOAD + gas_costs.G_BASE  # Warm SLOAD + POP
+    remaining_gas = gas_benchmark_value - totalgas
+    num_warm_loads = remaining_gas // (SPEEDUP * gas_increment)
+
+    # Build the complete code: warmup + repeated warm loads
+    sload_code = Op.SLOAD(0) + Op.POP if num_slots > 0 else Op.STOP
+    for i in range(1, num_slots):
+        sload_code = sload_code + Op.SLOAD(i) + Op.POP
```

Comment:
Idea: could use the storage value to point to the next storage slot to load. For instance this storage layout (key,value): `(0,1), (1,2), (2,0)` will thus iterate 0->1->2->0. Note: `POP` is thus not used here and neither is the `PUSH`). So this yields more SLOADS.